.github/workflows/ci.yml

```diff
@@ -134,3 +134,34 @@ jobs:
 
     - name: Check Formatting
       run: git diff --exit-code
+
+  tidy:
+    name: Analyzing on ${{ matrix.platform.name }}
+    runs-on: ${{ matrix.platform.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        platform:
+        - { name: Linux, os: ubuntu-latest }
+        - { name: macOS, os: macos-12 }
```

Comment:
Is there reason to assume that clang-tidy (a linter) will yield different results on different platforms?

Just trying to keep the number of overall CI jobs in reasonable bounds (this also applies to `format` etc).

Reply:
You have different code paths available on Linux vs macOS (vs Windows), so there will be differences

Reply:
That's right, different OSes compile different source files which means different source files get analyzed.

Reply:
Makes sense, thanks!